types/datum.go

```diff
@@ -1477,8 +1477,14 @@ func (d *Datum) toSignedInteger(sc *stmtctx.StatementContext, tp byte) (int64, e
 	case KindMysqlJSON:
 		return ConvertJSONToInt(sc, d.GetMysqlJSON(), false)
 	case KindBinaryLiteral, KindMysqlBit:
-		val, err := d.GetBinaryLiteral().ToInt(sc)
-		return int64(val), errors.Trace(err)
+		uval, err := d.GetBinaryLiteral().ToInt(sc)
+		val := int64(uval)
+		if err == nil {
+			if val > upperBound {
```

Comment:
need to check lowerboud overflow.

Reply:
the value is alwasy a positive number, so I think we don't need check the lowerBound